### PR TITLE
runtime: sleep 1 second after GetOOMEvent failed

### PR DESCRIPTION
**Pull request by liubin**
In some cases, for example agent crashed and not marked dead yet, the GetOOMEvent
will return errors like `connection reset by peer` or `ttrpc: closed`. Do a sleep
with 1 second (agent check interval) and let agent health check to do the check.

Fixes: #3064

Signed-off-by: bin liu <bin@hyper.sh>

### CODE REVIEW

**Review by devimc — APPROVED** [review]
thanks @liubin

File: containerd-shim-v2/wait.go
Comment (pmores) [inline containerd-shim-v2/wait.go:22]: @liubin Is this `defaultCheckInterval` by any chance related to the `defaultCheckInterval` in package virtcontainers (in `monitor.go`)?  If so, wouldn't it be more maintainable to reference the original value instead of redefining it here?

Comment (liubin) [reply containerd-shim-v2/wait.go]: Hi @pmores , you  are right, I have updated it.

Comment (pmores) [reply containerd-shim-v2/wait.go]: Thanks @liubin for the swift reaction!

Comment (fidencio) [reply containerd-shim-v2/wait.go]: Let me mark this one as resolved.

**Comment by fidencio** [comment]
- **jenkins-ci-Power8-ubuntu-18-04-initrd:**
```
Job for docker.service failed because the control process exited with error code.
See "systemctl status docker.service" and "journalctl -xe" for details.
Build step 'Execute shell' marked build as failure
```
@kata-containers/tests , who's the current contact for Power related issues?

**Comment by fidencio** [comment]
- **jenkins-ci-podman-clh-fedora-31:**
```
03:37:56 HEAD is now at 7dcd3aff build, release-notes.md: Document 0.11.0 release
03:37:56 ./scripts/dev_cli.sh: line 122: docker: command not found
03:37:56 [Cloud Hypervisor] Error pulling container image. Aborting.
```
Known issue and @jcvenegas is on this.
Wouldn't be worth to disable this CI till this get fixed? (considering that as this is a non required CI).

**Comment by fidencio** [comment]
- **jenkins-ci-ubuntu-18-04-initrd:**
```
time="2020-11-12T03:15:06.338140437Z" level=error msg="RunPodSandbox for &PodSandboxMetadata{Name:create-PodSandbox-for-container-41310e31-2495-11eb-b836-000d3ae40037,Uid:cri-test-uid41310e52-2495-11eb-b836-000d3ae40037,Namespace:cri-test-namespace41310e5a-2495-11eb-b836-000d3ae40037,Attempt:2,} failed, error" error="rpc error: code = Unavailable desc = failed to create containerd task: transport is closing: unavailable"
time="2020-11-12T03:15:06.342539677Z" level=info msg="StopPodSandbox for \"\""
time="2020-11-12T03:15:06.342643278Z" level=error msg="StopPodSandbox for \"\" failed" error="an error occurred when try to find sandbox \"\": Prefix can't be empty"
time="2020-11-12T03:15:06.343102482Z" level=info msg="RemovePodSandbox for \"\""
time="2020-11-12T03:15:06.343174683Z" level=error msg="RemovePodSandbox for \"\" failed" error="an error occurred when try to find sandbox \"\": Prefix can't be empty"
time="2020-11-12T03:15:06.365478390Z" level=error msg="RunPodSandbox for &PodSandboxMetadata{Name:create-PodSandbox-for-container-413d3188-2495-11eb-85f8-000d3ae40037,Uid:cri-test-uid413d31c5-2495-11eb-85f8-000d3ae40037,Namespace:cri-test-namespace413d31d0-2495-11eb-85f8-000d3ae40037,Attempt:2,} failed, error" error="rpc error: code = Unavailable desc = failed to create containerd task: transport is closing: unavailable"
time="2020-11-12T03:15:06.368504018Z" level=info msg="StopPodSandbox for \"\""
time="2020-11-12T03:15:06.368592618Z" level=error msg="StopPodSandbox for \"\" failed" error="an error occurred when try to find sandbox \"\": Prefix can't be empty"
time="2020-11-12T03:15:06.368985122Z" level=info msg="RemovePodSandbox for \"\""
time="2020-11-12T03:15:06.369051423Z" level=error msg="RemovePodSandbox for \"\" failed" error="an error occurred when try to find sandbox \"\": Prefix can't be empty"
time="2020-11-12T03:15:07.300094539Z" level=error msg="RunPodSandbox for &PodSandboxMetadata{Name:create-PodSandbox-for-container-41365457-2495-11eb-a433-000d3ae40037,Uid:cri-test-uid4136547b-2495-11eb-a433-000d3ae40037,Namespace:cri-test-namespace41365486-2495-11eb-a433-000d3ae40037,Attempt:2,} failed, error" error="rpc error: code = Unavailable desc = failed to create containerd task: transport is closing: unavailable"
time="2020-11-12T03:15:07.302390560Z" level=info msg="StopPodSandbox for \"\""
time="2020-11-12T03:15:07.302476161Z" level=error msg="StopPodSandbox for \"\" failed" error="an error occurred when try to find sandbox \"\": Prefix can't be empty"
time="2020-11-12T03:15:07.302806864Z" level=info msg="RemovePodSandbox for \"\""
time="2020-11-12T03:15:07.302859164Z" level=error msg="RemovePodSandbox for \"\" failed" error="an error occurred when try to find sandbox \"\": Prefix can't be empty"
time="2020-11-12T03:15:07.328907205Z" level=error msg="RunPodSandbox for &PodSandboxMetadata{Name:create-PodSandbox-for-container-4139061b-2495-11eb-bdd8-000d3ae40037,Uid:cri-test-uid4139063a-2495-11eb-bdd8-000d3ae40037,Namespace:cri-test-namespace41390643-2495-11eb-bdd8-000d3ae40037,Attempt:2,} failed, error" error="rpc error: code = Unavailable desc = failed to create containerd task: transport is closing: unavailable"
time="2020-11-12T03:15:07.331705831Z" level=info msg="StopPodSandbox for \"\""
time="2020-11-12T03:15:07.331796532Z" level=error msg="StopPodSandbox for \"\" failed" error="an error occurred when try to find sandbox \"\": Prefix can't be empty"
time="2020-11-12T03:15:07.332222936Z" level=info msg="RemovePodSandbox for \"\""
time="2020-11-12T03:15:07.332287936Z" level=error msg="RemovePodSandbox for \"\" failed" error="an error occurred when try to find sandbox \"\": Prefix can't be empty"
time="2020-11-12T03:15:07.349356894Z" level=error msg="RunPodSandbox for &PodSandboxMetadata{Name:create-PodSandbox-for-container-41348207-2495-11eb-946a-000d3ae40037,Uid:cri-test-uid4134822c-2495-11eb-946a-000d3ae40037,Namespace:cri-test-namespace41348239-2495-11eb-946a-000d3ae40037,Attempt:2,} failed, error" error="rpc error: code = Unavailable desc = failed to create containerd task: transport is closing: unavailable"
time="2020-11-12T03:15:07.354359041Z" level=info msg="StopPodSandbox for \"\""
time="2020-11-12T03:15:07.354444341Z" level=error msg="StopPodSandbox for \"\" failed" error="an error occurred when try to find sandbox \"\": Prefix can't be empty"
time="2020-11-12T03:15:07.354876345Z" level=info msg="RemovePodSandbox for \"\""
time="2020-11-12T03:15:07.354933846Z" level=error msg="RemovePodSandbox for \"\" failed" error="an error occurred when try to find sandbox \"\": Prefix can't be empty"
time="2020-11-12T03:15:07.369552781Z" level=debug msg="received signal" signal=terminated
time="2020-11-12T03:15:07.369711283Z" level=info msg="Stop CRI service"
time="2020-11-12T03:15:07.369855184Z" level=info msg="Stop CRI service"
```
@liubin, can this be related to your PR?
@kata-containers/tests, @kata-containers/runtime, have you seen this happening before?

**Comment by fidencio** [comment]
- **jenkins-ci-ubuntu-1804-cloud-hypervisor-k8s-e2e-containerd-minimal:**
@likebreath, @jcvenegas, may I ask you guys to take a look if this is a known error? Also, if possible, what's the error? I couldn't figure it out by a quick look there.

**Comment by fidencio** [comment]
- **jenkins-metrics-ubuntu-virtiofs-18-04:**:
```
4:00:38 
04:00:38 Report Summary:
04:00:38 +-----+----------------------+---------------------+---------------+--------+-------+--------+--------+-------+------+-----+
04:00:38 | P/F |         NAME         |         FLR         |     MEAN      |  CEIL  |  GAP  |  MIN   |  MAX   |  RNG  | COV  | ITS |
04:00:38 +-----+----------------------+---------------------+---------------+--------+-------+--------+--------+-------+------+-----+
04:00:38 | P   | boot-times           | 95.0%               | 98.4%         | 105.0% | 10.0% | 91.7%  | 118.1% | 28.7% | 5.4% |  20 |
04:00:38 | *F* | memory-footprint     | Failed to load JSON | exit status 2 |        |       |        |        |       |      |     |
04:00:38 | P   | memory-footprint-ksm | 95.0%               | 100.0%        | 105.0% | 10.0% | 100.0% | 100.0% | 0.0%  | 0.0% |   1 |
04:00:38 | P   | blogbench            | 90.0%               | 96.0%         | 110.0% | 20.0% | 96.0%  | 96.0%  | 0.0%  | 0.0% |   1 |
04:00:38 | P   | blogbench            | 90.0%               | 94.5%         | 110.0% | 20.0% | 94.5%  | 94.5%  | 0.0%  | 0.0% |   1 |
04:00:38 +-----+----------------------+---------------------+---------------+--------+-------+--------+--------+-------+------+-----+
04:00:38 Fails: 1, Passes 4
04:00:38 Failed
04:00:38 checkmetrics FAILED (1)
```

Seems that we don't fail because the memory footprint went up, but because we couldn't read the JSON file. Interesting.
@kata-containers/tests, @kata-containers/runtime, have you ever faced this before?

**Review by fidencio — APPROVED** [review]
@liubin, thanks, lgtm (and I'm also forwarding @pmores's ack!) :-)

Please, let's wait till we get an answer about the CI failures before we get this merged.

**Comment by likebreath** [comment]
* jenkins-metrics-ubuntu-virtiofs-18-04
It is failing on the error `| *F* | memory-footprint     | Failed to load JSON | exit status 2 |  ` which I encountered quite many times. I learnt the workaround is simply restarting the CI. (I just restarted the CI here). Ping @GabyCT in case she got a better solution/fix.

* jenkins-ci-ubuntu-1804-cloud-hypervisor-k8s-e2e-containerd-minimal
This is a strange failure, as I believe the tests are essentially passing. I compared this failing one job [186](http://jenkins.katacontainers.io/job/kata-containers-runtime-ubuntu-PR-cloud-hypeprvisor-k8s-e2e-containerd-minimal/186/console) and the most recent successful job [188](http://jenkins.katacontainers.io/job/kata-containers-runtime-ubuntu-PR-cloud-hypeprvisor-k8s-e2e-containerd-minimal/188/console). The only major difference I can find is that the failing job is not reporting `.ci/teardown.sh "$WORKSPACE/artifacts"` while the successful job did. Also, note that both jobs are complaining some non-critical errors, e.g. `No such file or directory`, `Permission denied`, `ERROR: kata-log-parser`, which I believe are not the reason for the failure. 

Again, trying the old tricks and restarted the CI. @jcvenegas any insights?

**Comment by likebreath** [comment]
Also, put DNM label to prevent merging the PR by accident before clarifying CI failures. Feel free to remove the label whenever you find appropriate.

**Comment by likebreath** [comment]
Note that: the error on `jenkins-ci-ubuntu-1804-cloud-hypervisor-k8s-e2e-containerd-minimal` is resolved and I believe the CI is flaky. Anyway, we got one less CI failure. Are we waiting for someone's inputs on the failures from `jenkins-ci-Power8-ubuntu-18-04-initrd ` and `jenkins-ci-ubuntu-18-04-initrd `?

**Comment by fidencio** [comment]
@likebreath, I don't think it makes sense to keep blocking the PRs for the CIs that are not officially required.
I've removed the `do-not-merge` label.  Feel free to merge the patches and I'll open issues (Tomorrow) so we can keep tracking the failures.

Where would be the best place to open the issues? Here or at the tests repo?

**Comment by likebreath** [comment]
> I've removed the `do-not-merge` label. Feel free to merge the patches and I'll open issues (Tomorrow) so we can keep tracking the failures.
> 
> Where would be the best place to open the issues? Here or at the tests repo?

@fidencio Make senses to me. I'd assume the tests repo is a better for CI related issues. Merging the PR now. Thanks a lot.

**Pull request merged**
